listenbrainz/webserver/static/js/src/recommendations/Recommendations.tsx

```diff
@@ -62,6 +65,76 @@ export default class Recommendations extends React.Component<
     );
   }
 
+  componentDidMount(): void {
+    const { user, currentUser } = this.props;
+    const { currRecPage } = this.state;
+    if (currentUser?.name === user?.name) {
+      this.loadFeedback();
+    }
+    window.history.replaceState(null, "", `?page=${currRecPage}`);
```

Comment:
The browser history part is not fully working; I'll note it down here for posterity but I think it should be fixed in another PR.
I don't mind doing that PR if you want.

`currRecPage` is set in the state in the constructor to `1`, so if I visit the recs page with `?page=2` in my URL (suppose for example I want to send you the second page of my recs), I'll land on page 1 regardless of my URL parameter.

Reply:
Yes, we have a ticket to implement that i.e. take `page` arg from URL and jump to that page, I think we need to slice the `recommendations` prop using `page` and update the page to that. Will do in a separate PR. 

> I don't think using the browser back button to go to the next page is working either.

But I am not sure what is causing this and how to fix this? Is there a fix in RecentListens for this?

Reply:
Yep! https://github.com/metabrainz/listenbrainz-server/blob/master/listenbrainz/webserver/static/js/src/RecentListens.tsx#L115-L116

The `popstate` event triggers when the browser history changes (for example clicking the browser's 'previous' button).

The `handleURLChange` mechanism will not be 100% the same, because in RecentListens each time we call the api to load listens, while in recs page the recommendations will already be there in props and we just need to set a new slice in the state instead, but that gives you an idea of the implementation